90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -622,24 +641,30 @@ draw_diff() {
     return
   fi
 
-  diff_target="${snapshot%%@*}"
-  zdebug "base filesystem: ${diff_target}"
+  base_fs="${snapshot%%@*}"
+  zdebug "base filesystem: ${base_fs}"
 
-  CLEAR_SCREEN=1 load_key "${diff_target}"
+  CLEAR_SCREEN=1 load_key "${base_fs}"
 
-  if ! mnt="$( mount_zfs "${diff_target}" )" ; then
-    zerror "unable to mount ${diff_target}"
+  if ! mnt="$( mount_zfs "${base_fs}" )" ; then
+    zerror "unable to mount ${base_fs}"
     return
   fi
 
+  zdebug "executing: zfs diff -F -H ${snapshot} ${diff_target}"
   coproc zfs_diff ( zfs diff -F -H "${snapshot}" "${diff_target}" )
 
   # Bash won't use an FD referenced in a variable on the left side of a pipe
   exec 3>&"${zfs_diff[0]}"
 
   # shellcheck disable=SC2154
+  line_one="$( center_string "---${snapshot}" )"
+  left_pad="${line_one//---${snapshot}/}"
+  line_one="$( colorize red "${line_one}" )"
+  line_two="${left_pad}$( colorize green "+++${diff_target}" )"
+
   sed "s,${mnt},," <&3 | HELP_SECTION=diff-viewer ${FUZZYSEL} --prompt "${snapshot} > " \
```

Comment:
Since the preview now shows diff-like headers that provide necessary context, I think the prompt here should be something generic (maybe just a `>`). Using `${snapshot}` is a bit confusing since the diff contents really applies to both sides.

Reply:
Agreed. Chagned!